### PR TITLE
Refactor Tradeparams to prevent unnecessary renders.

### PR DESCRIPTION
**Pull request by easteregg**

### CODE REVIEW

File: src/javascript/app_2/Modules/Trading/Components/Form/TradeParams/Duration/duration-wrapper.jsx [hunk 1]
```diff
@@ -19,10 +18,10 @@ class DurationWrapper extends React.Component {
         const new_duration_unit  = this.props.duration_units_list[0].value;
         const new_duration_value = this.props.getDurationFromUnit(new_duration_unit);
 
-        this.props.onChangeUiStore({ name: `${this.props.is_advanced_duration ? 'advanced' : 'simple'}_duration_unit`, value: new_duration_unit });
+        this.props.onChangeUiStore({ name: `${this.props.is_advanced_duration ? 'advanced' : 'simple'}_duration_unit`, value: +new_duration_unit });
```
Comment (OskarAhl) [inline src/javascript/app_2/Modules/Trading/Components/Form/TradeParams/Duration/duration-wrapper.jsx]: duration_unit is a string?

File: src/javascript/app_2/Modules/Trading/Components/Form/TradeParams/amount.jsx [hunk 2]
```diff
@@ -130,4 +133,20 @@ Amount.propTypes = {
     validation_errors : PropTypes.object,
 };
 
-export default observer(Amount);
+export default connect(({ modules, client }) => ({
+    // Amount component props
```
Comment (OskarAhl) [inline src/javascript/app_2/Modules/Trading/Components/Form/TradeParams/amount.jsx]: nit: this comment seems redundant?

File: src/javascript/app_2/Modules/Trading/Components/Form/TradeParams/amount.jsx [hunk 3]
```diff
@@ -1,109 +1,112 @@
...
-            is_incrementable
-            is_nativepicker={is_nativepicker}
-            is_negative_disabled
-            max_length={10}
-            name='amount'
-            onChange={onChange}
-            type='tel'
-            value={amount}
-        />;
+class Amount extends React.Component {
```
[the remark this answers is not in the document]
Comment (easteregg) [reply src/javascript/app_2/Modules/Trading/Components/Form/TradeParams/amount.jsx]: I did use `trace` and other stuff to track re-rendering, forgot to revert this back to functional component. I make it functional one more time.

File: src/javascript/app_2/Modules/Trading/Components/Form/TradeParams/amount.jsx (hunk 2 above)
Comment (easteregg) [reply src/javascript/app_2/Modules/Trading/Components/Form/TradeParams/amount.jsx]: Done.

**Comment by negarn** [comment]
@msamprz @easteregg conflict

**Comment by easteregg** [comment]
@negarn Resolved.

**Pull request merged**